contracts/test/integration/IntegrationTestMakerRouter.t.sol

```diff
@@ -0,0 +1,415 @@
+//SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.13;
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {DSTest} from "../unit/utils/DSTest.sol";
+import {Vm} from "../unit/utils/Vm.sol";
+import {MainnetAddresses} from "./fixtures/MainnetAddresses.sol";
+import {IDSSPSM} from "../../pcv/maker/IDSSPSM.sol";
+import {MakerRouter} from "../../pcv/maker/MakerRouter.sol";
+import {PegStabilityModule} from "../../peg/PegStabilityModule.sol";
+import {IVolt} from "../../volt/IVolt.sol";
+import {Core} from "../../core/Core.sol";
+import {Constants} from "../../Constants.sol";
+
+contract IntegrationTestMakerRouter is DSTest {
+    using SafeCast for *;
+
+    MakerRouter private makerRouter;
+
+    Core private core = Core(MainnetAddresses.CORE);
+    IDSSPSM private daiPSM = IDSSPSM(MainnetAddresses.MAKER_DAI_USDC_PSM);
+    PegStabilityModule private feiPSM =
+        PegStabilityModule(MainnetAddresses.FEI_DAI_PSM);
+
+    IERC20 private dai = IERC20(MainnetAddresses.DAI);
+    IVolt private fei = IVolt(MainnetAddresses.FEI);
+    IERC20 private usdc = IERC20(MainnetAddresses.USDC);
+
+    Vm public constant vm = Vm(HEVM_ADDRESS);
+
+    uint256 public constant USDC_SCALING_FACTOR = 1e12;
+
+    uint256 public constant mintAmount = 200_000_000e18;
+
+    function setUp() public {
+        makerRouter = MakerRouter(MainnetAddresses.MAKER_ROUTER);
+
+        fei.approve(address(makerRouter), type(uint256).max);
+        dai.approve(address(makerRouter), type(uint256).max);
+
+        vm.prank(MainnetAddresses.DAI_USDC_USDT_CURVE_POOL);
+        dai.transfer(address(feiPSM), mintAmount + 50_000_000e18);
+
+        vm.prank(MainnetAddresses.FEI_GOVERNOR);
+        fei.mint(address(this), mintAmount);
+    }
+
+    function testSwapFeiForDai(uint64 amountFeiIn) public {
```

Comment:
maybe fuzz with `uint96` because `uint64` will fuzz up to 18 FEI maximum (2**64/1e18)